Aidan/dockerfileless mode (#232)
latchbio/latch: docs/source/subcommands.md

$ latch init PACKAGE_NAME
```

> This command is used to initialize a new workflow directory, using one of four templates. It will create a directory with the name `PACKAGE_NAME` and populate it with the necessary files to get started.

Reviewer: ```suggestion
This command is used to initialize a new workflow root by populating it with the required boilerplate. Optionally, a template workflow can be used as a starting point.
```

> The parameter `PACKAGE_NAME` is the name of the directory that `latch init` will create and populate with example files.

Reviewer: ```suggestion
`PACKAGE_NAME` is the name of the target directory. It will be created if it does not exist.
```

### Options

#### `--template`, `-t`

> One of `r`, `conda`, `subprocess`, `empty`. If not provided, user will be prompted for input.

Reviewer: explain the templates here

#### `--dockerfile`, `-d`

> Generate a Dockerfile for the workflow.

Reviewer: ```suggestion
Generate a Dockerfile for the workflow instead of relying on [auto-generation.](LINK TO DOCKERFILE AUTOGEN PAGE)
```

> #### `--cuda`

Reviewer: these really need to be one parameter like `--base-image`

Reviewer: also templates AND base images is a bit redundant, need a way to explain the difference and decide what cause where

This command can be used to generate an example workflow ([this workflow](https://genomebiology.biomedcentral.com/track/pdf/10.1186/gb-2009-10-3-r25.pdf), to be specific) for reference purposes. It has all of the [required elements](basics/what_is_a_workflow.md) for a workflow, and can also be used as boilerplate - simply replace the logic with your logic, adding more tasks as necessary.

The parameter `PACKAGE_NAME` is the name of the directory that `latch init` will create and populate with example files.

### Options

> > `--template`, `-t`
> >
> > One of `r`, `conda`, `subprocess`, `empty`. If not provided, user will be prompted for input.
> >
> > `--dockerfile`, `-d`
> >
> > Generate a Dockerfile for the workflow.

Reviewer 2: ```suggestion
#### `--template`, `-t`

One of `r`, `conda`, `subprocess`, `empty`. If not provided, user will be prompted for input.

#### `--dockerfile`, `-d`

Generate a Dockerfile for the workflow.
```

The first argument specifies the local path in which to look for workflow objects. Inside the local path should be a single directory called `wf` containing `__init__.py`, any helper python files, a `Dockerfile`, and a `version` files. See [here](basics/what_is_a_workflow.md) for more info.

### Options

> > `--disable-auto-version`, `-d`

Reviewer 2: same as above, convert to subheadings etc.

The first argument specifies the local path in which to look for workflow objects. Inside the local path should be a single directory called `wf` containing `__init__.py`, any helper python files, a `Dockerfile`, and a `version` files. See [here](basics/what_is_a_workflow.md) for more info.

### Options

#### `--disable-auto-version`, `-d`

> Whether to automatically bump the version of the workflow each time register is called.

Reviewer: ```suggestion
Do not include the workflow contents hash in the workflow version. The `version` must be manually updated after each registration.
```
+ explain why this might be useful

Reviewer: or delete if not useful lol

#### `--remote`, `-r`

> Use a remote server to build workflow.

Reviewer: ```suggestion
Use a remote server to build the workflow.
```

## `latch dockerfile`

> This command generates a Dockerfile for a workflow using files in the workflow directory.

Reviewer: ```suggestion
Generate a `Dockerfile` using files in the specified workflow root.
```

### Options

> > `--name`

Reviewer 2: same as above

### Options

#### `--name`

> If provided, refines the output to instead list out all available versions of the workflow with the specified name

Reviewer: ```suggestion
If provided, list only the versions with the specified name
```

Reviewer: why is this not the default wot
